### ISSUE TITLE
Paused video starts playing when seek is executed

### ISSUE THREAD

**Issue by stereodenis** [TheWidlarzGroup/react-native-video#2775]
# Bug

I have several videos in ScrollView. When I tried to rewind video to start via seek -- it starts to play

## Platform
* iOS
* Android

## Environment info

React native info output:

```bash
System:
    OS: macOS 12.4
    CPU: (10) arm64 Apple M1 Pro
    Memory: 189.64 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.5.0 - /opt/homebrew/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 8.12.1 - /opt/homebrew/bin/npm
    Watchman: 2022.07.04.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK: Not Found
  IDEs:
    Android Studio: 2021.2 AI-212.5712.43.2112.8609683
    Xcode: 13.4.1/13F100 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.15 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.68.2 => 0.68.2 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

Library version: 5.2.0

## Steps To Reproduce

1. Seek paused video
2. It starts to play

## Expected behaviour

1. Seek paused video
2. It not starts to play

## Reproducible sample code
```
<ScrollView
          ref={scrollviewRef}
          horizontal
          pagingEnabled
          bounces={false}
          onMomentumScrollEnd={onMomentumScrollEnd}
          onLayout={handleOnLayout}
          contentOffset={{
            x: initialPageNumber * SCREEN_WIDTH,
            y: 0,
          }}
        >
          {videos.map((video, index) => (
<Video
          ref={onVideoRef(index)}
          repeat
          source={{
            uri: video.link,
          }}
          resizeMode={"cover"}
          paused={isPaused || !isCurrentVideo}
          style={styles.video}
          poster={poster}
          posterResizeMode={"cover"}
        />
)}
</ScrollView>
```

**Comment by eodonohoe** [comment]
I had a similar issue, but it turned out that it was due to my paused attribute being a state variable that would update asynchronously, where as the seek method seems to runs immediately. 

 I was able to get it to work by calling seek in a useEffect hook that passed in the paused variable as a dependency.

**Comment by CreeJee** [comment]
I don't know this patch is right way but it works in my case
here's share my patch

>version: 6.0.0-alpha.4

```diff
diff --git a/node_modules/react-native-video/android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java b/node_modules/react-native-video/android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java
index 8850610..c26d437 100644
--- a/node_modules/react-native-video/android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java
+++ b/node_modules/react-native-video/android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java
@@ -1708,6 +1708,7 @@ class ReactExoplayerView extends FrameLayout implements
     public void seekTo(long positionMs) {
         if (player != null) {
             player.seekTo(positionMs);
+            setPlayWhenReady(!isPaused);
             eventEmitter.seek(player.getCurrentPosition(), positionMs);
         }
     }

```

**Comment by freeboub** [comment]
@CreeJee You patch looks correct, but I don't reproduce the issue on basic sample on master (6.0.0-alpha.pre5).
As provided sample is inside a scrollView, can it be an issue with it ?

**Comment by finnp** [comment]
I had this issue on 5.2.1 (iOS) and it was fixed through updating to 6.0.0-alpha.5.

**Comment by freeboub** [comment]
let's close this issue as not reproduced on V6.0.0
